### PR TITLE
Get events

### PR DESCRIPTION
**Pull request by fang-helen**
Created a servlet to get events as a JSON string -- either a user's saved or created events while logged in, or a list of all events while not logged in. 
Also made modifications to the event creation servlet so that events can only be created when logged in, and saves the creator id of the logged in user when doing so.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/UserServlet.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      Key userKey = KeyFactory.createKey("User", userEmail);
+      try {
+        Entity userEntity = datastore.get(userKey);
+        List<Entity> results = new ArrayList<>();
+        if(request.getParameter("get")!= null && request.getParameter("get").equals("saved")) {
```
Comment (Yunni) [inline src/main/java/com/google/sps/servlets/UserServlet.java:45]: Consider use switch case here like
```
switch (request.getParameter("get")) {
  case "saved": handleSaved(); break;
  case "created": handleCreated(); break;
  default: // throw exception
}
private void handleSaved() {
....
```
This pattern will be easier to expand in the future.

File: src/test/java/com/google/sps/UserServletTest.java [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    goalEntity.setProperty("eventDescription", "We're cleaning up the lake");
+    goalEntity.setProperty("streetAddress", "678 Lakeview Way");
+    goalEntity.setProperty("city", "Lakeside");
+    goalEntity.setProperty("state", "Michigan");
+    goalEntity.setProperty("date", "2020-17-05");
+    goalEntity.setProperty("startTime", "14:00");
+    goalEntity.setProperty("endTime", "");
+    goalEntity.setProperty("coverPhoto", "");
+    goalEntity.setProperty("tags", "['environment']");
+    goalEntity.setProperty("creator", "test@example.com");
```
Comment (tc466) [inline src/test/java/com/google/sps/UserServletTest.java:214]: Would be good to refactor into a helper method. This way, the method can be reused in the next test below.

Comment (fang-helen) [reply src/test/java/com/google/sps/UserServletTest.java]: Done!

File: src/main/java/com/google/sps/servlets/UserServlet.java (hunk 1 above)
Comment (fang-helen) [reply src/main/java/com/google/sps/servlets/UserServlet.java:45]: Makes sense, thanks!

**Pull request merged**